### ISSUE TITLE
Tensor Programs V: Tuning Large Neural Networks via Zero-Shot Hyperparameter Transfer

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#216]
# TL; DR

- mu-parameterization enables zero-shot hyper-parameter transfers, which saves effort and resources traditionally devoted to HP tuning on large models.

# Motivation

- With Maximal Update Parameterization (mu-parameterization), the optimal HP remains stable across training scales (width, depth, batch size, etc).

# Formulation

![image](https://github.com/reyllama/paper-reviews/assets/46308407/d3dc0668-c920-43dd-b109-52f953e2f7e9)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/c2852490-22d5-4b29-b44b-e6534f464828)

- Standard parameterization is inspired by CLT, but it becomes unstable after parameter updates.
- The authors (and the authors of the original mu-p paper) show that standard parameterization can expose the model to activation explosion after a single step of SGD update.
- mu-p, on the other hand, scales the logits to remain bounded, making them better transferrable to wider (and bigger) models.
- Hence, one can simply copy&paste optimal HPs found from HP search on small networks to larger training setting.

# Evaluation

![image](https://github.com/reyllama/paper-reviews/assets/46308407/2ea1b01f-93aa-4b50-af09-5f985680f2e7)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/644adbcb-cbef-45b2-b0dc-d7808436f0ed)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/d54b4c83-5fbb-48c3-8242-0564b6372f64)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/b30be8ca-bc3e-4812-9715-5d78a7ffaa73)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/aed2c74c-c3bf-4daf-b433-5e983777cde4)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/387b9f43-7366-47cb-af0a-104445146fcf)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/7bdfcf53-afe8-4e42-938e-1271102ad566)
![image](https://github.com/reyllama/paper-reviews/assets/46308407/12fea46e-56ba-4726-a10e-fc6c9193b35e)

# Inspiration

- Great work with huge practical implications.
- Personally, have always thought that theoretical works concerning infinite-width neural networks more or less lack practicality (because of the very idea of infinite-width), but obviously I was wrong and the power of **principled engineering** can be riveting.
